Bookish/Bookish.DataAccess/LibraryService.cs

```diff
@@ -32,29 +32,25 @@ public Book GetCopies(string isbn)
                 "LEFT JOIN Users ON Users.userId=Loans.userId " +
                 $"WHERE LibraryBooks.isbn='{isbn}';";
 
-            var bookDictionary = new Dictionary<string, Book>();
-
+            var outBook = new Book();
 
             var bookWithCopies = databaseConnection.Query<Book, BookCopy, Book>(
                     query,
                     (book, bookCopy) =>
                     {
-                        Book bookEntry;
-
-                        if (!bookDictionary.TryGetValue(book.isbn, out bookEntry))
+                        if (book.isbn != outBook.isbn)
                         {
-                            bookEntry = book;
-                            bookEntry.bookCopies = new List<BookCopy>();
-                            bookDictionary.Add(bookEntry.isbn, bookEntry);
+                            outBook = book;
+                            outBook.bookCopies = new List<BookCopy>();
                         }
 
-                        bookEntry.bookCopies.Add(bookCopy);
-                        return bookEntry;
+                        outBook.bookCopies.Add(bookCopy);
+                        return outBook;
                     },
                     splitOn: "bookId")
-                .Distinct()
-                .ToList();
-            return bookWithCopies.Single();
+                .FirstOrDefault();
+                
+            return bookWithCopies ?? new Book();
```

Comment:
Empty book doesn't seem like a great default. I think you should make the return type nullable and make sure you handle null in the controller 🙂